### ISSUE TITLE
I think there are some mistakes

### ISSUE THREAD

**Issue by William20234** [luuuyi/CBAM.PyTorch#18]
After applying  Channel Attention Module, maybe it would be better to apply a convolution layer in order to modify the channels to the original value (usually 3 channels), instead of applying Spatial Attention Module instantly. Or Spatial Attention Module can't make sense.

**Comment by William20234** [comment]
My Advice:
class Bottleneck(nn.Module):
    

    def __init__(self, inplanes, planes, stride=1, downsample=None,expansion=4):
        super(Bottleneck, self).__init__()
        self.conv1 = nn.Conv2d(inplanes, planes, kernel_size=1, bias=False)
        self.bn1 = nn.BatchNorm2d(planes)
        self.conv2 = nn.Conv2d(planes, planes, kernel_size=3, stride=stride,
                               padding=1, bias=False)
        self.bn2 = nn.BatchNorm2d(planes)
        self.conv3 = nn.Conv2d(planes, planes * expansion, kernel_size=1, bias=False)
        self.bn3 = nn.BatchNorm2d(planes * expansion)
        self.relu = nn.ReLU(inplace=True)

        self.conv4=nn.Conv2d(planes * expansion, inplanes, kernel_size=1, bias=False)

        self.ca = ChannelAttention(planes * expansion)
        self.sa = SpatialAttention()

        self.downsample = downsample
        self.stride = stride

    def forward(self, x):
        residual = x

        out = self.conv1(x)
        out = self.bn1(out)
        out = self.relu(out)

        out = self.conv2(out)
        out = self.bn2(out)
        out = self.relu(out)

        out = self.conv3(out)
        out = self.bn3(out)

        out = self.ca(out) * out
        out=self.conv4(out)
        out = self.sa(out) * out

        if self.downsample is not None:
            residual = self.downsample(x)
        
        out += residual

        out = self.relu(out)

        return out
